include/albatross/src/models/sparse_gp.hpp

```diff
@@ -620,6 +625,82 @@ class SparseGaussianProcessRegression
   GrouperFunction independent_group_function_;
 };
 
+// rebase_inducing_points takes a Sparse GP which was fit using some set of
+// inducing points and creates a new fit relative to new inducing points.
+// Note that this will NOT be the equivalent to having fit the model with
+// the new inducing points since some information may have been lost in
+// the process.
+template <typename ModelType, typename FeatureType, typename NewFeatureType>
+auto rebase_inducing_points(
+    const FitModel<ModelType, Fit<SparseGPFit<FeatureType>>> &fit_model,
+    const std::vector<NewFeatureType> &new_inducing_points) {
+
+  FitModel<ModelType, Fit<SparseGPFit<NewFeatureType>>> output(
+      fit_model.get_model(), Fit<SparseGPFit<NewFeatureType>>());
+  Fit<SparseGPFit<NewFeatureType>> &new_fit = output.get_fit();
+
+  new_fit.train_features = new_inducing_points;
+
+  const Eigen::MatrixXd K_zz =
+      fit_model.get_model().get_covariance()(new_inducing_points);
+  new_fit.train_covariance = Eigen::SerializableLDLT(K_zz);
+
+  const JointDistribution new_prediction =
+      fit_model.predict(new_inducing_points).joint();
+  new_fit.information = new_fit.train_covariance.solve(new_prediction.mean);
+
+  // Here P is the posterior covariance at the new inducing points.  If
+  // we consider the case where we rebase and then use the resulting fit
+  // to predict the new inducing points then we see that the predictive
+  // covariance (see documentation above) would be,:
+  //
+  //    C = K_zz - Q_zz + K_zz Sigma K_*zz
+  //
+  // We can use this, knowing that at the inducing points K_zz = Q_zz, to
+  // derive our updated Sigma,
+  //
+  //    C = K_zz - K_zz + K_zz Sigma K_**
```

Comment:
Why does `K_*zz` become `K_**`? Something to do with being at the induction points?

Reply:
Ah the last `K_**` is a typo.  I use `*` to indicate something you're predicting and in this case I'm predicting at the new inducing points `z`.